shakenfist/daemons/queues.py

```diff
@@ -277,9 +277,7 @@ def instance_delete(instance_uuid, new_state, error_msg=None):
         instance = virt.Instance.from_db(instance_uuid)
         if instance:
             instance.delete()
-            instance.update_instance_state(new_state)
-            if error_msg:
-                instance.update_error_message(error_msg)
+            instance.update_instance_state(new_state, error_message=error_msg)
```

Comment:
I left a lot of inefficient function names because I didn't the want the code change to be too massive.